alphagov/notifications-api
# Add letters pdf task & api

## What

- Added a task to create a cmyk pdf and store this in the letters-pdf bucket on s3.
- Only allow this for services that have `letters_as_pdf` set
- Ensure that for these services the `build_dvla_file` is not called so that DVLA only get pdfs or csv files to process.
- Ensure that letter notifications API calls the create letters pdf task when in `letters_as_pdf` service
- Make sure that only notifications not in `letters_as_pdf` services are processed for api calls that are building dvla files.

## How to manually test

- From admin settings, switch a service to send letters as pdf
- then from a letter template upload a csv file
  - you should be able to see the newly created CMYK files on the s3 bucket for that environment
  - or from celery logs you should be able to see something like this -
  `PDF Letter [noti id] reference [reference] created at [datetime], [pdf size] bytes`
  `Uploading letters [pdf file location] to [environment]-letters-pdf`

- From an API call - 
`PYTHONPATH=. python ./utils/make_api_call.py http://localhost:6011 [test key for service with letters_as_pdf] create --type=letter --template=[letter template id] --personalisation="{\"address_line_1\":\"someone\", \"address_line_2\":\"London\", \"postcode\":\"N1\"}" --reference="test letters"`

## Reference 

Step 4/5 https://www.pivotaltracker.com/story/show/151511579
